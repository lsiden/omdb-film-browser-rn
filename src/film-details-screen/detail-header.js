import React from "react"
import PropTypes from "prop-types"
import { connect } from "react-redux"
import { Text, View } from "react-native"

import { detailStyles as style } from "styles"
import Filler from "components/filler"

const titleWrapperStyle = { flexDirection: "row", flexWrap: "wrap" }

export const detailHeader = ({ filmDetails }) => {
  return (
    <View style={style.titleWrapper}>
      <View style={titleWrapperStyle}>
        <Text style={style.title}>{filmDetails.Title}</Text>
        <Text style={style.year}>{`  (${filmDetails.Year})`}</Text>
      </View>
      <Filler />
    </View>
  )
}

detailHeader.propTypes = {
  filmDetails: PropTypes.object.isRequired
}

export default connect(state => ({
  filmDetails: state.filmDetails
}))(detailHeader)
