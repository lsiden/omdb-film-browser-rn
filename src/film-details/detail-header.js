import React from "react"
import PropTypes from "prop-types"
import { Text, View } from "react-native"
import { connect } from "react-redux"

import { detailStyles } from "styles"
import BackButton from "./back-button"

export const detailHeader = props => {
  const { filmDetails } = props
  return (
    <View style={detailStyles.header}>
      <BackButton />
      <View style={detailStyles.titleWrapper}>
        <Text style={detailStyles.title}>{filmDetails.Title}</Text>
        <Text style={detailStyles.year}>{`  (${filmDetails.Year})`}</Text>
      </View>
    </View>
  )
}

detailHeader.propTypes = {
  filmDetails: PropTypes.object.isRequired,
}

export default connect(state => ({
  filmDetails: state.filmDetails,
}))(detailHeader)
