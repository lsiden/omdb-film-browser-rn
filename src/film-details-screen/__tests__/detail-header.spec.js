import React from "react"
import renderer from "react-native-test-utils"

import { detailHeader as DetailHeader } from "../detail-header"

const filmDetails = require("__test__/__fixture__/film-details.json")

const defaultProps = () => {
  return {
    filmDetails,
    navigation: {
      goBack: () => {}
    }
  }
}

const createWrapper = (props = {}) =>
  renderer(<DetailHeader {...{ ...defaultProps(), ...props }} />)

it("renders", () => {
  const wrapper = createWrapper()
  expect(wrapper.toJSON()).toMatchSnapshot()
})
