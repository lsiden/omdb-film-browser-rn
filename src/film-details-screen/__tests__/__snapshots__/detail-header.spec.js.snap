// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<View
  style={
    Object {
      "alignItems": "baseline",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "space-between",
      "marginBottom": 8,
      "marginTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "flexWrap": "wrap",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginLeft": 10,
        }
      }
    >
      The Battle of Algiers
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "lightgray",
          "fontSize": 16,
        }
      }
    >
        (1966)
    </Text>
  </View>
  <View
    style={
      Object {
        "width": 24,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "transparent",
        }
      }
    >
      x
    </Text>
  </View>
</View>
`;
